packages/core-database/lib/wallet-manager.js

```diff
@@ -376,6 +376,41 @@ module.exports = class WalletManager {
     return data
   }
 
+  /**
+   * Determine the list of active delegates for the given round.
+   * @param  {Number} maxDelegates
+   * @return {Array}
+   */
+  determineActiveDelegates (maxDelegates) {
+    const mapWallets = wallets => wallets.map(wallet => ({
+      publicKey: wallet.vote || wallet.publicKey,
+      balance: wallet.balance ? wallet.balance.toFixed() : 0
+    }))
+
+    let delegates = mapWallets(this.allByPublicKey().filter(wallet => !!wallet.vote))
+
+    // NOTE: At the launch of the blockchain we may not have enough delegates.
+    // In order to have enough forging delegates we complete the list in a
+    // deterministic way (alphabetical order of publicKey).
+    if (delegates.length < maxDelegates) {
+      const publicKeys = delegates.map(delegate => delegate.publicKey)
+
+      let placeholders = this.allByUsername()
+
+      if (publicKeys.length) {
+        placeholders = placeholders.filter(wallet => publicKeys.includes(wallet.publicKey))
+      }
+
+      placeholders = placeholders.sort((a, b) => a.publicKey - b.publicKey)
+
+      delegates = delegates.concat(mapWallets(placeholders))
+    }
+
+    return delegates
+      .sort((a, b) => b.balance - a.balance) // TODO: improve sorting here to avoid collision of equal vote balances
```

Comment:
Shouldn't this use the `voteBalance` property instead of `balance`? It looks like this is sorting according to delegate's own balance, not according to the sum of all voters' balances.